scylla/src/statement/prepared_statement.rs

```diff
@@ -513,13 +513,14 @@ impl From<SerializationError> for PartitionKeyError {
     }
 }
 
-pub(crate) type PartitionKeyValue<'ps> = (&'ps [u8], &'ps ColumnSpec);
+pub(crate) type PartitionKeyValue<'ps, 'spec> = (&'ps [u8], &'ps ColumnSpec<'spec>);
 
-pub(crate) struct PartitionKey<'ps> {
-    pk_values: SmallVec<[Option<PartitionKeyValue<'ps>>; PartitionKey::SMALLVEC_ON_STACK_SIZE]>,
+pub(crate) struct PartitionKey<'ps, 'spec: 'ps> {
+    pk_values:
+        SmallVec<[Option<PartitionKeyValue<'ps, 'spec>>; PartitionKey::SMALLVEC_ON_STACK_SIZE]>,
```

Comment:
Is there a reason for having 2 lifetimes? I think this type is covariant over both 'ps and 'spec, so it should be fine to unify those lifetimes - but maybe I'm just talking gibberish, idk. @piodul ?